src/rngs/os.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Interface to the random number generator of the operating system.
+
+use getrandom::getrandom;
+use rand_core::{CryptoRng, RngCore, Error, ErrorKind, impls};
+
+/// A random number generator that retrieves randomness from from the
+/// operating system.
+///
+/// This is a zero-sized struct. It can be freely constructed with `OsRng`.
+/// 
+/// The implementation is provided by the [getrandom] crate. Refer to
+/// [getrandom] documentation for details.
+///
+/// [getrandom]: https://crates.io/crates/getrandom
+#[derive(Clone, Copy, Debug)]
+pub struct OsRng;
```

Comment:
Add `Default` impl?

Reply:
Why?

The only use I could see is if you wanted generic code over `R: Rng + Default`, which is a very strange bound.

Reply:
It's a Clippy recommendation, but only for types with a `new()` method: https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default

Reply:
I know that, but considering the old `new` constructor returned a `Result` it did not make sense. I don't believe it makes any more sense now.

Reply:
One use-case for `Default` impl is an ability to abstract over both `OsRng` and `ThreadRng`.

Reply:
Fair enough. Done (turns out you can derive `Default` in this case).